alpha/apps/kaltura/modules/extwidget/actions/serveFlavorAction.class.php

```diff
@@ -915,22 +920,35 @@ protected function getLiveParams()
 	}
 
 	/**
-	 * @param array $flavors
+	 * @param array $assets
 	 * @return array
 	 */
-	public static function buildSequencesArray($flavors)
+	public static function buildSequencesArray($assets)
 	{
 		$sequences = array();
 
-		foreach ($flavors as $flavor)
+		foreach ($assets as $asset)
 		{
-			$syncKey = $flavor->getSyncKey(flavorAsset::FILE_SYNC_FLAVOR_ASSET_SUB_TYPE_ASSET);
-			list ($file_sync, $path, $sourceType) = kFileSyncUtils::getFileSyncServeFlavorFields($syncKey, $flavor, self::getPreferredStorageProfileId(), self::getFallbackStorageProfileId());
+			$syncKey = $asset->getSyncKey(flavorAsset::FILE_SYNC_FLAVOR_ASSET_SUB_TYPE_ASSET);
+			list ($file_sync, $path, $sourceType) = kFileSyncUtils::getFileSyncServeFlavorFields($syncKey, $asset, self::getPreferredStorageProfileId(), self::getFallbackStorageProfileId());
 			if(!$path)
 			{
-				KalturaLog::debug('missing path for flavor ' . $flavor->getId() . ' version ' . $flavor->getVersion());
+				KalturaLog::debug('missing path for asset ' . $asset->getId() . ' version ' . $asset->getVersion());
+			}
+			$sequence = array('clips' => array(self::getClipData($path, $asset, $sourceType)));
+			if (is_callable(array($asset, 'getLanguage')) && $asset->getLanguage())
+			{
+				$languageCode = languageCodeManager::getLanguageCode($asset->getLanguage(), true);
```

Comment:
This is a wrong approach, the correct way is to add all asset tree the getLanguage function, where who ever did not implement it will return null, while others will follow this logic.